src/Compilers/Core/Portable/CodeGen/ILBuilderEmit.cs

```diff
@@ -110,6 +110,18 @@ internal void EmitArrayBlockInitializer(ImmutableArray<byte> data, SyntaxNode sy
             EmitToken(initializeArray, syntaxNode, diagnostics);
         }
 
+        internal void EmitCreateSpan(ImmutableArray<byte> data, Cci.IMethodReference createSpanHelper, SyntaxNode syntaxNode, DiagnosticBag diagnostics)
+        {
+            // map a field to the block (that makes it addressable via a token)
+            var field = module.GetFieldForData(data, alignment: 1, syntaxNode, diagnostics);
```

Comment:
>alignment: 1 [](http://example.com/codeflow?start=53&length=12)

Why is this the right value?

Reply:
> and I think the same reasoning should apply here as well.

This logic has a big flaw . We are doing different things and requirements are different. We cannot simply rely on what ```EmitArrayBlockInitializer``` is doing because ```RuntimeHelpers.InitializeArray``` doesn't have any alignment requirements, but ```CreateSpan``` does. And that, actually can be observed at other places where compiler uses ```CreateSpan```.

Reply:
Thanks. I see we are aligning to the byte size where we use `CreateSpan` today, which makes sense, I overlooked that previously.